scripts/pylib/twister/twisterlib.py

```diff
@@ -2094,6 +2094,9 @@ def run_cmake_script(args=[]):
         p = subprocess.Popen(cmd, **kwargs)
         out, _ = p.communicate()
 
+        ansi_escape = re.compile(r'\x1B(?:[@-Z\\-_]|\[[0-?]*[ -/]*[@-~])')
```

Comment:
would below be better?
```
escapes = ''.join([chr(char) for char in range(1, 32)])
translator = str.maketrans('', '', escapes)
t = s.translate(translator)
```

Reply:
Unfortunately not, because then the ANSI escape sequences would be left over. out contained in the problematic case the following string:
b'\x1b[0m{"ZEPHYR_TOOLCHAIN_VARIANT" : "zephyr", "SDK_VERSION": "0.13.0", "ZEPHYR_SDK_INSTALL_DIR" : "/opt/zephyr-sdk-0.13.0"}\x1b[0m\n'
With your proposal the "reset all modes" sequence (0m) would be left over. To be honest, I have stolen the regex from stackoverflow, but it looks reasonable to me: https://stackoverflow.com/questions/14693701/how-can-i-remove-the-ansi-escape-sequences-from-a-string-in-python

Reply:
so you issue is not to remove the escape chars but the control chars from a SSH console. Do you ever try to not parser the ssh cosnole output direct instead you can use tee to redirect your console output to a file?

Reply:
Actually there is no ssh-console in my case involved, it is just a plain call to twister.py from inside a makefile. I'm not yet really sure where these things are coming from exactly, but I experienced this problem only after an update to the most recent zephyr-repo revision.
Funny thing, the problem doesn't occur if I call twister directly, without make. But assuming I won't be the only one with such a problem and considering that ANSI escape sequences are really of no use here I am proposing this solution. Probably easier for other people as well.

I would actually prefer your approach over mine, as it is more readable. But as soon as all these possible combinations of escape sequences come in to play it is probably better to solve it with a regex.

Reply:
would this possible to share your makefile? Just do not want to confuse people when they see such regex in code.

Reply:
Sure, this is a minimal failing-as-expected example:

```
#!/usr/bin/make -f

all: 
	rm -fR build/test
	./zephyr/scripts/twister --overflow-as-errors --clobber-output --outdir build/test -T test/ --platform native_posix --platform qemu_cortex_r5

.PHONY: all
```

This then fails with a json.decoder.JSONDecodeError, due to the ANSI escape characters.

Reply:
so you put this file as a shell script?
test.sh
```
#!/usr/bin/make -f

all: 
	rm -fR build/test
	./zephyr/scripts/twister --overflow-as-errors --clobber-output --outdir build/test -T test/ --platform native_posix --platform qemu_cortex_r5

.PHONY: all
```
and you run it with 
```
bash ./test.sh
```
what shell are you using?

Reply:
No, this is the content of Makefile and I execute make all in a bash.